src/main/java/eu/dissco/backend/repository/MongoRepository.java

```diff
@@ -0,0 +1,45 @@
+package eu.dissco.backend.repository;
+
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.Projections.include;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.mongodb.client.MongoDatabase;
+import eu.dissco.backend.exceptions.NotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@RequiredArgsConstructor
+public class MongoRepository {
+
+  private final MongoDatabase database;
+  private final ObjectMapper mapper;
+
+  public JsonNode getVersion(String id, int version, String collectionName)
```

Comment:
Small thing: consider calling this getByVersion? You're not retrieving the version, you're using the version to retrieve a record. 
I don't see any uses for this yet

Reply:
Hmm it is used by three services, SpecimenService, AnnotationService and DigitalMediaService. Will change to getByVersion